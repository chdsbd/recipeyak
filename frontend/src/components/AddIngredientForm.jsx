import React from 'react'

import { ButtonPrimary } from './Buttons'

const AddIngredientForm = ({
  handleAddIngredient,
  cancelAddIngredient,
  handleInputChange,
  quantity,
  name,
  description,
  optional,
  loading,
  error,
}) =>
  <form onSubmit={ async e => {
    e.preventDefault()
    await handleAddIngredient()
    document.querySelector('#firstinput').focus()
  }
  }>
  <div className="add-ingredient-grid mb-2 mt-2">

    <div>
      <input
        id="firstinput"
        onChange={ handleInputChange }
        onFocus={ e => e.target.select() }
        autoFocus
        value={ quantity }
        className={ 'my-input' + (error ? ' is-danger' : '') }
        type="text"
        placeholder="3 lbs"
        name="quantity"/>
    </div>

    <div>
      <input
        onChange={ handleInputChange }
        onFocus={ e => e.target.select() }
        value={ name }
        className={ 'my-input' + (error ? ' is-danger' : '') }
        type="text"
        placeholder="tomato"
        name="name"/>
    </div>

    <div className="grid-entire-row">
      <input
        onChange={ handleInputChange }
        onFocus={ e => e.target.select() }
        value={ description }
        className={ 'my-input' + (error ? ' is-danger' : '') }
        type="text"
        placeholder="diced at 3cm"
        name="description"/>
      { error
          ? <p className="fs-4 c-danger">A recipe needs at least one ingredient</p>
          : null
      }
    </div>
  </div>

  <label className="d-flex align-items-center cursor-pointer mb-2">
  <input
    onChange={ handleInputChange }
    checked={ optional }
    name="optional"
    type='checkbox'
    className="mr-2"/>
    Optional
  </label>

  <div className="field is-grouped">
    <p className="control">
      <ButtonPrimary
        disabled={ quantity === '' && name === '' }
        className="is-small"
        type="submit"
        name="add ingredient"
        loading={ loading }>
        Add
      </ButtonPrimary>
    </p>
    <p className="control">
      <input
        onClick={ cancelAddIngredient }
        className="my-button is-small"
        type="button"
        name="cancel add ingredient"
        value="cancel"/>
    </p>
  </div>
</form>

export default AddIngredientForm
