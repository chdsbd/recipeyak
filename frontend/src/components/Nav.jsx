import React from "react";
import { Link } from "react-router-dom";

import NavLink from "../containers/NavLink";
import Logo from "./Logo";
import Dropdown from "./Dropdown";
import NotificationsDropdown from "./NotificationsDropdown";
import UserDropdown from "./UserDropdown";

import { teamURL } from "../urls";

const Teams = ({ teams, loading }) => {
  if (loading) {
    return <p>Loading...</p>;
  }

  if (teams.length === 0) {
    return <p className="text-muted fs-3 align-self-center">No teams.</p>;
  }

  return (
    <div className="text-left">
      {teams.map(({ id, name }) => (
        <p key={id}>
          <NavLink to={teamURL(id, name)} activeClassName="fw-500">
            {name}
          </NavLink>
        </p>
      ))}
    </div>
  );
};

class Navbar extends React.Component {
  static defaultProps = {
    teams: [],
    scheduleURL: ""
  };

  componentWillMount() {
    this.props.fetchData();
  }

  render() {
    const {
      avatarURL,
      loggedIn = true,
      logout,
      email,
      loggingOut,
      className = "",
      toggleDarkMode,
      darkMode,
      scheduleURL
    } = this.props;

    const buttons = loggedIn ? (
      <div className="d-flex align-center p-relative justify-content-center flex-wrap">
        <NavLink
          to="/recipes/add"
          activeClassName="active"
          className="better-nav-item"
        >
          Add
        </NavLink>

        <NotificationsDropdown />

        <Dropdown name="Teams" relative={false}>
          <Teams loading={this.props.loadingTeams} teams={this.props.teams} />
          <Link to="/t/create" className="mt-1 ">
            Create a Team
          </Link>
        </Dropdown>

        <NavLink
          to="/recipes"
          activeClassName="active"
          className="better-nav-item"
        >
          Browse
        </NavLink>
        <NavLink
          to={scheduleURL}
          activeClassName="active"
          className="better-nav-item"
        >
          Schedule
        </NavLink>

        <UserDropdown
          avatarURL={avatarURL}
          email={email}
          toggleDarkMode={toggleDarkMode}
          darkMode={darkMode}
          logout={logout}
          loggingOut={loggingOut}
        />
      </div>
    ) : (
      <div className="d-flex hide-sm">
        <NavLink to="/login" className="better-nav-item">
          Login
        </NavLink>
        <NavLink to="/signup" className="better-nav-item">
          Signup
        </NavLink>
      </div>
    );

    return (
      <nav className={`nav flex-wrap ${className}`}>
        <Link
          to="/"
          className="better-nav-item pl-0 pr-0 fs-2rem fw-normal font-family-title"
        >
          <Logo width="45px" />
          <span>Recipe Yak</span>
        </Link>
        {buttons}
      </nav>
    );
  }
}

export default Navbar;
