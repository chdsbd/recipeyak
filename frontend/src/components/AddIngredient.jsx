import React from 'react'
import AddIngredientForm from './AddIngredientForm'

class AddIngredient extends React.Component {
  constructor (props) {
    super(props)

    this.emptyState = {
      addingIngredient: false,
      quantity: '',
      unit: '',
      name: '',
      description: ''
    }

    this.state = this.emptyState
  }

  handleInputChange = e => {
    this.setState({ [e.target.name]: e.target.value })
  }

  handleAddIngredient = e => {
    e.preventDefault()
    this.addIngredient(this.props.id, this.state.ingredient)
  }

  addingIngredient = () => {
    this.setState({ addingIngredient: true })
  }

  cancelAddIngredient = () => {
    this.setState(this.emptyState)
  }

  clearInputs = () => {
    this.setState({ quantity: '', unit: '', name: '', description: '' })
  }

  addIngredient = async (id, ingredient) => {
    // TODO: pass this prop in from Recipe or via store
    const { quantity, unit, name, description } = this.state
    await this.props.addIngredient(id, { quantity, unit, name, description })
    this.clearInputs()
  }

  render () {
    const { state } = this

    const units = [
      'large',
      'small',
      'medium',
      'tsp',
      'tbsp',
      'cup',
      'lb',
      'kg',
      'mg',
      'gram',
      'liter',
      'ml',
      'gal.',
      'oz',
      'fl oz',
      'pint',
      'quart',
      'pinch',
      'inch',
      'cm',
      'm',
      'mm'
    ]

    return (
      state.addingIngredient
      ? <AddIngredientForm
          handleAddIngredient={this.handleAddIngredient}
          cancelAddIngredient={this.cancelAddIngredient}
          handleInputChange={this.handleInputChange}
          units={units}
          quantity={this.state.quantity}
          unit={this.state.unit}
          name={this.state.name}
          description={this.state.description}
        />
      : <p className="flex-center">
        <button
          onClick={ this.addingIngredient }
          className="button is-link">
          Add another
        </button>
      </p>
    )
  }
}

export default AddIngredient
