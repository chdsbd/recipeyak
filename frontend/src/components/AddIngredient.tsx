import React from "react"
import PropTypes from "prop-types"
import AddIngredientForm from "./AddIngredientForm"

export interface IIngredientBasic {
  quantity: string
  name: string
  description: string
}

interface IAddIngredientProps {
  readonly onCancel: () => void
  readonly id: number
  readonly addIngredient: (
    id: number,
    { quantity, name, description }: IIngredientBasic
  ) => void
  readonly loading: boolean
  readonly autoFocus: boolean
}

interface IAddIngredientState {
  readonly quantity: string
  readonly name: string
  readonly description: string
  readonly optional: boolean
  readonly addingIngredient: boolean
}

export type AddIngredientFields = Exclude<
  keyof IAddIngredientState,
  "addingIngredient"
>

export default class AddIngredient extends React.Component<
  IAddIngredientProps,
  IAddIngredientState
> {
  static propTypes = {
    id: PropTypes.number.isRequired,
    addIngredient: PropTypes.func.isRequired,
    loading: PropTypes.bool.isRequired,
    autoFocus: PropTypes.bool,
    onCancel: PropTypes.func.isRequired
  }

  static defaultProps = {
    loading: false
  }

  emptyState: IAddIngredientState = {
    quantity: "",
    name: "",
    description: "",
    optional: false,
    addingIngredient: false
  }

  state: IAddIngredientState = this.emptyState

  handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    this.setState(({
      [e.target.name]: e.target.value
    } as unknown) as IAddIngredientState)
  }

  addingIngredient = () => {
    this.setState({ addingIngredient: true })
  }

  cancelAddIngredient = () => {
    this.props.onCancel()
    this.setState(this.emptyState)
  }

  clearInputs = () => {
    this.setState(this.emptyState)
  }

  render() {
    const { clearInputs, cancelAddIngredient, handleInputChange } = this
    const { id, addIngredient, loading, autoFocus } = this.props
    const { quantity, name, description, optional } = this.state

    return (
      <AddIngredientForm
        handleAddIngredient={async () => {
          await addIngredient(id, { quantity, name, description })
          clearInputs()
        }}
        loading={loading}
        cancelAddIngredient={cancelAddIngredient}
        handleInputChange={handleInputChange}
        quantity={quantity}
        name={name}
        description={description}
        optional={optional}
        autoFocus={autoFocus}
      />
    )
  }
}
