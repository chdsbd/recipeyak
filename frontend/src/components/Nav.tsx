import React from "react"
import {
  Button,
  Menu,
  MenuItem,
  MenuItemProps,
  MenuTrigger,
  Popover,
  Separator,
} from "react-aria-components"
import {
  Configure,
  InstantSearch,
  useHits,
  useSearchBox,
} from "react-instantsearch"
import { Link, useHistory } from "react-router-dom"
import useOnClickOutside from "use-onclickoutside"

import { useIsLoggedIn } from "@/auth"
import { clx } from "@/classnames"
import { Avatar } from "@/components/Avatar"
import { SearchInput } from "@/components/Forms"
import Logo from "@/components/Logo"
import { NavLink } from "@/components/Routing"
import { SearchResult } from "@/components/SearchResult"
import {
  pathHome,
  pathLogin,
  pathProfileById,
  pathRecipeAdd,
  pathRecipeDetail,
  pathRecipesList,
  pathSchedule,
  pathSettings,
  pathSignup,
  pathTeamList,
} from "@/paths"
import { useAuthLogout } from "@/queries/authLogout"
import { useTeam } from "@/queries/teamFetch"
import { useSearchClient } from "@/queries/useSearchClient"
import { useGlobalEvent } from "@/useGlobalEvent"
import { useTeamId } from "@/useTeamId"
import { useUser } from "@/useUser"

function UserDropdown() {
  const user = useUser()

  const logoutUser = useAuthLogout()

  const menuItems: Array<
    | { type: "menuitem"; label: string; to: string; onClick?: undefined }
    | { type: "menuitem"; label: string; to?: undefined; onClick: () => void }
    | { type: "separator"; id: string }
  > = [
    {
      type: "menuitem",
      label: "Profile",
      to: pathProfileById({ userId: String(user.id) }),
    },
    {
      type: "menuitem",
      label: "Settings",
      to: pathSettings({}),
    },
    {
      type: "menuitem",
      label: "Teams",
      to: pathTeamList({}),
    },
    {
      type: "separator",
      id: "separator-1",
    },
    {
      type: "menuitem",
      label: "Logout",
      onClick: () => {
        logoutUser.mutate()
      },
    },
  ]

  const teamId = useTeamId()
  const team = useTeam({ teamId })

  return (
    <MenuTrigger>
      <Button className="flex cursor-pointer items-center justify-center rounded-full border-none bg-[unset] p-0 focus-visible:outline focus-visible:outline-[3px] focus-visible:-outline-offset-2 focus-visible:outline-[rgb(47,129,247)]">
        <Avatar avatarURL={user.avatarURL} />
      </Button>
      <Popover className="w-56 origin-top-left overflow-auto rounded-md border border-solid border-[var(--color-border)] bg-[var(--color-background-calendar-day)] p-2 shadow-lg outline-none">
        <Menu
          className="outline-none"
          onAction={(key) => {
            const metadata = menuItems.find(
              (x) => x.type === "menuitem" && x.label === key,
            )
            if (metadata?.type === "menuitem") {
              metadata.onClick?.()
            }
          }}
          disabledKeys={["meta-info"]}
        >
          <MenuItem id="meta-info" className="pl-2">
            <div className="pb-1 ">
              <span className="">{user.name ?? user.email}</span>
              <span> Â· </span>
              <span className="text-sm ">{team.data?.name}</span>
              <div className="text-sm ">{user.email}</div>
            </div>
          </MenuItem>
          <Separator className="my-1 h-[1px] bg-[var(--color-border)]" />
          {menuItems.map((menuItem) => {
            if (menuItem.type === "separator") {
              return (
                <Separator
                  id={menuItem.id}
                  key={menuItem.id}
                  className="my-1 h-[1px] bg-[var(--color-border)]"
                />
              )
            }
            return (
              <ActionItem
                id={menuItem.label}
                key={menuItem.label}
                href={menuItem.to}
              >
                {menuItem.label}
              </ActionItem>
            )
          })}
        </Menu>
      </Popover>
    </MenuTrigger>
  )
}

function ActionItem(props: Omit<MenuItemProps, "className">) {
  return (
    <MenuItem
      {...props}
      className={
        "flex cursor-pointer rounded px-2 py-1 [transition:background_.12s_ease-out] hover:bg-[var(--color-border)] focus-visible:outline-[3px] focus-visible:-outline-offset-2 focus-visible:outline-[rgb(47,129,247)] "
      }
    />
  )
}

function WordMark() {
  return <span className="hidden text-2xl sm:block">Recipe Yak</span>
}

function AuthButtons() {
  return (
    <div className="flex justify-self-end">
      <NavLink className={navItemCss} to={pathLogin({})}>
        Login
      </NavLink>
      <NavLink className={navItemCss} to={pathSignup({})}>
        Signup
      </NavLink>
    </div>
  )
}

function NavButtons() {
  return (
    <div className="relative flex items-center justify-center gap-2 justify-self-end">
      <div className="flex print:!hidden sm:gap-2">
        <NavLink
          to={pathRecipeAdd({})}
          className={navItemCss}
          activeClassName={activeNavItemCss}
        >
          Add
        </NavLink>
        <NavLink
          to={pathRecipesList({})}
          className={navItemCss}
          activeClassName={activeNavItemCss}
        >
          Browse
        </NavLink>
        <NavLink
          to={pathSchedule({})}
          className={navItemCss}
          activeClassName={activeNavItemCss}
        >
          Calendar
        </NavLink>
      </div>

      <UserDropdown />
    </div>
  )
}

const navItemCss =
  "flex shrink-0 grow-0 cursor-pointer items-center justify-center rounded-md px-2 py-1 text-[14px] font-medium leading-[1.5] text-[var(--color-text)] transition-all [transition:background_.12s_ease-out] hover:bg-[var(--color-background-calendar-day)] hover:text-[var(--color-link-hover)] active:bg-[var(--color-border)]"
const activeNavItemCss = "bg-[var(--color-background-calendar-day)]"

function isInputFocused() {
  const activeElement = document.activeElement
  return (
    activeElement !== document.body &&
    activeElement !== null &&
    (activeElement.tagName === "INPUT" || activeElement.tagName === "TEXTAREA")
  )
}

/**
 *
 * Implementation is very similar to "Search" in UserHome.tsx.
 */
function Search() {
  const history = useHistory()
  const { query, refine } = useSearchBox()
  const { hits } = useHits<{
    readonly id: number
    readonly name: string
    readonly archived_at: string | null
    readonly author: string | null
  }>()

  // If a user clicks outside of the dropdown, we want to hide the dropdown, but
  // keep their search query.
  //
  // The alternative would be to clear the search query when clicking outside,
  // but I'm not sure that's desirable.
  const [isClosed, setIsClosed] = React.useState(false)
  const searchInputRef = React.useRef<HTMLInputElement>(null)

  const ref = React.useRef(null)
  useOnClickOutside(ref, () => {
    setIsClosed(true)
  })

  useGlobalEvent({
    keyDown(e) {
      if (
        (e.key === "k" && e.metaKey) ||
        (e.key === "/" && !isInputFocused())
      ) {
        searchInputRef.current?.focus()
        e.preventDefault()
      }
    },
  })

  const resetForm = () => {
    refine("")
    setIsClosed(false)
  }

  const handleSearchKeydown = (e: React.KeyboardEvent<HTMLInputElement>) => {
    // We need to extract the key from the synthetic event before we lose the
    // event.
    const key = e.key
    const suggestion = hits[0]
    if (!suggestion) {
      return
    }
    if (key === "Enter") {
      resetForm()
      history.push(pathRecipeDetail({ recipeId: suggestion.id.toString() }))
    }
  }

  return (
    <div ref={ref} className="flex w-full">
      <SearchInput
        ref={searchInputRef}
        value={query}
        placeholder="Press / to search"
        onChange={(e) => {
          refine(e.target.value)
        }}
        onKeyDown={handleSearchKeydown}
        onFocus={() => {
          setIsClosed(false)
        }}
      />
      {query && !isClosed && (
        <div className="absolute inset-x-0 top-[60px] z-10 w-full sm:inset-x-[unset] sm:max-w-[400px]">
          <SearchResult
            hits={hits}
            onClick={() => {
              resetForm()
            }}
          />
        </div>
      )}
    </div>
  )
}

export function Navbar({ includeSearch = true }: { includeSearch?: boolean }) {
  const isLoggedIn = useIsLoggedIn()
  const teamId = useTeamId()
  const searchClient = useSearchClient()
  return (
    <nav className="flex h-[3.5rem] shrink-0 justify-between gap-1 px-3 pb-1 print:!hidden md:grid md:grid-cols-3">
      <div className="flex items-center justify-start gap-2">
        <Link
          to={pathHome({})}
          className={
            "flex shrink-0 grow-0 cursor-pointer items-center justify-center rounded-md"
          }
        >
          <Logo width="40px" />
        </Link>
        <Link to={pathHome({})} className={clx(navItemCss, "hidden sm:block")}>
          {isLoggedIn ? (
            <span className="font-medium ">Home</span>
          ) : (
            <WordMark />
          )}
        </Link>
      </div>
      <div className="flex grow items-center">
        {includeSearch && searchClient && (
          <InstantSearch
            searchClient={searchClient}
            indexName="recipes"
            // use custom routing config so we can have `search` be our query parameter.
            // https://www.algolia.com/doc/guides/building-search-ui/going-further/routing-urls/react-hooks/
            routing={{
              stateMapping: {
                stateToRoute(uiState) {
                  const indexUiState = uiState["recipes"]
                  return { search: indexUiState.query }
                },
                routeToState(routeState) {
                  return {
                    ["recipes"]: {
                      query: routeState.search,
                    },
                  }
                },
              },
            }}
          >
            <Search />
            <Configure hitsPerPage={1000} filters={`team_id:${teamId}`} />
          </InstantSearch>
        )}
      </div>
      {isLoggedIn ? <NavButtons /> : <AuthButtons />}
    </nav>
  )
}
