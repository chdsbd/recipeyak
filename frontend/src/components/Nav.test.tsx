import React from "react"
import { Navbar } from "@/components/Nav"
import { TestProvider } from "@/testUtils"
import renderer from "react-test-renderer"
import { createEmptyStore } from "@/store/store"

describe("<Nav/>", () => {
  it("renders default", () => {
    const tree = renderer.create(
      <TestProvider>
        <Navbar />
      </TestProvider>
    )
    expect(tree).toMatchSnapshot()
  })

  test("render when logged in", () => {
    const emptyState = createEmptyStore().getState()
    const store = createEmptyStore({
      ...emptyState,
      user: {
        ...emptyState.user,
        loggedIn: true
      }
    })
    const tree = renderer.create(
      <TestProvider store={store}>
        <Navbar />
      </TestProvider>
    )
    expect(tree).toMatchSnapshot()
  })
})
