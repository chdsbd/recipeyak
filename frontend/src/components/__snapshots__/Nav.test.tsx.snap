// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Nav/> render when logged in 1`] = `
.c2 {
  position: relative;
}

.c5 {
  position: absolute;
  left: auto;
  right: 0;
  z-index: 1000;
  padding: 0.5rem;
  margin: 0.125rem 0 0;
  font-size: 1rem;
  white-space: nowrap;
  background-color: #f9f9f9;
  border: 1px solid rgba(0,0,0,0.15);
  border-radius: 0.25rem;
  display: none;
}

.c1 {
  font-size: 1.5rem;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c3 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  font-size: 1rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  line-height: 1.5;
  padding: 0.5rem 0.75rem;
  font-weight: 500;
  color: #4a4a4a;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 0.1s;
  transition: all 0.1s;
  display: none;
}

.c3:hover {
  color: #707070;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c3:active {
  -webkit-transform: translateY(1px);
  -ms-transform: translateY(1px);
  transform: translateY(1px);
}

.c3.active {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c0 {
  -webkit-flex-wrap: 1;
  -ms-flex-wrap: 1;
  flex-wrap: 1;
  margin-bottom: 0.25rem;
  padding-left: 0.75rem;
  padding-right: 0.75rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

@media (max-width:630px) {
  .c1 {
    display: none;
  }
}

@media (max-width:630px) {
  .c4 {
    z-index: 1000;
    display: none;
    position: absolute;
    background-color: white;
    border: 1px solid lightgray;
    top: 100%;
    right: 0;
  }
}

@media (max-width:630px) {
  .c3 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

<nav
  className="c0"
>
  <a
    className="better-nav-item pb-1 pt-1 pl-0 pr-0 fw-normal"
    href="/"
    onClick={[Function]}
  >
    <svg
      className="mr-1 fill-text-color "
      version="1.1"
      viewBox="0 0 512 512"
      width="40px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M436.6,87.8c-0.9,0.1-1.9,2.7-3.7,7.9c-2.8,8.1-4.3,14.3-5.2,18.3c-1.4,5.6-3,12.1-5.2,18.8c-2.7,7.9-4.2,12.5-7.7,18.2
        c-4,6.3-8.2,10.6-10.6,13c-2.7,2.7-8,7.5-15.8,12.3c-4.9,3-12.4,6.9-22.3,10c-8.9,2.7-16.9,4.6-24.2,5.9c-2.8,0.5-7.5,1.3-16.4,2.3
        c-5.3,0.6-19.8,2.1-43.9,2.9c-6.7,0.2-15.4,0.4-25.6,0.5c-10.2,0-18.9-0.3-25.6-0.5c-24-0.8-38.5-2.3-43.9-2.9
        c-8.9-1-13.5-1.7-16.4-2.3c-7.3-1.3-15.3-3.2-24.2-5.9c-9.9-3-17.3-6.9-22.3-10c-7.8-4.8-13.1-9.6-15.8-12.3
        c-2.4-2.4-6.6-6.6-10.6-13c-3.6-5.7-5.1-10.3-7.7-18.2c-2.3-6.6-3.8-13.2-5.2-18.8c-1-4-2.4-10.2-5.2-18.3c-1.8-5.2-2.8-7.9-3.7-7.9
        c-4.1-0.3-13.2,23.2-12.1,47.4c0.3,7.5,1.4,25.9,14.4,44.9c10.2,14.8,23.4,23.8,31.6,29.4c10.6,7.2,14.1,7.4,19.7,11.5
        c12.4,9.2,14.8,20.5,22.4,60c6,30.9,11.1,64,22,95.4c2.5,7.3,6.6,27.5,22.5,40.9c7.4,6.2,16.6,10.8,16.6,10.8
        c3.2,1.6,11.6,5.5,23.4,7.8c7.4,1.4,14.1,1.9,20.1,1.8c5.9,0.1,12.7-0.4,20.1-1.8c11.8-2.3,20.2-6.2,23.4-7.8c0,0,9.2-4.6,16.6-10.8
        c15.9-13.4,20-33.6,22.5-40.9c10.9-31.4,16-64.5,22-95.4c7.6-39.6,10.1-50.9,22.4-60c5.6-4.1,9.1-4.3,19.7-11.5
        c8.2-5.6,21.4-14.5,31.6-29.4c13-19,14.1-37.4,14.4-44.9C449.7,110.9,440.6,87.5,436.6,87.8z M333.2,258.7
        c-7.1,53.7-13.8,82-24.6,107.8c-0.4,1.1-1.3,3.3-2.9,6.1c-5.1,9.2-13.1,24.4-31,30.8c-18.3,6.6-36.3,0.3-37.3,0
        c-17.9-6.4-25.9-21.6-31-30.8c-1.6-2.8-2.5-5.1-2.9-6.1c-10.8-25.7-17.5-54.1-24.6-107.8c-1.3-10.1-2.3-18,3.7-22.9
        c6.5-5.3,15.5-4,38.4-3.2c20.4,0.7,49.6,0.7,70,0c23-0.8,31.9-2.1,38.4,3.2C335.4,240.7,334.5,248.6,333.2,258.7z"
      />
    </svg>
    <span
      className="c1"
    >
      Recipe Yak
    </span>
  </a>
  <div
    className="d-flex align-center p-relative justify-content-center flex-wrap"
  >
    <div
      className="c2"
    >
      <a
        className="c3"
        onClick={[Function]}
      >
        Menu 
        <svg
          className="fill-text-color"
          style={
            Object {
              "width": "1.5em",
            }
          }
          viewBox="0 0 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <path
              d="M14.1711599,9.3535 L9.99925636,13.529 L5.82735283,9.3535 C5.51262415,9.0385 5.73543207,8.5 6.18054835,8.5 L13.8179644,8.5 C14.2630807,8.5 14.4858886,9.0385 14.1711599,9.3535"
              fill="inherit"
            />
          </g>
        </svg>
      </a>
      <div
        className="c4"
      >
        <a
          className="better-nav-item "
          href="/recipes/add"
          onClick={[Function]}
        >
          Add
        </a>
        <div
          className="c2"
        >
          <a
            className="better-nav-item"
            onClick={[Function]}
            tabIndex={0}
          >
            <span>
              Teams
            </span>
            <svg
              className="fill-text-color"
              style={
                Object {
                  "width": "1.5em",
                }
              }
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g>
                <path
                  d="M14.1711599,9.3535 L9.99925636,13.529 L5.82735283,9.3535 C5.51262415,9.0385 5.73543207,8.5 6.18054835,8.5 L13.8179644,8.5 C14.2630807,8.5 14.4858886,9.0385 14.1711599,9.3535"
                  fill="inherit"
                />
              </g>
            </svg>
          </a>
          <div
            className="c5"
          >
            <p>
              Loading...
            </p>
            <a
              className="mt-1 "
              href="/t/create"
              onClick={[Function]}
            >
              Create a Team
            </a>
          </div>
        </div>
        <div
          className="c2"
        >
          <a
            className="better-nav-item"
            onClick={[Function]}
            tabIndex={0}
          >
            <span>
              Notifications
            </span>
            <svg
              className="fill-text-color"
              style={
                Object {
                  "width": "1.5em",
                }
              }
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g>
                <path
                  d="M14.1711599,9.3535 L9.99925636,13.529 L5.82735283,9.3535 C5.51262415,9.0385 5.73543207,8.5 6.18054835,8.5 L13.8179644,8.5 C14.2630807,8.5 14.4858886,9.0385 14.1711599,9.3535"
                  fill="inherit"
                />
              </g>
            </svg>
          </a>
          <div
            className="c5"
          >
            <p
              className="text-muted text-small align-self-center"
            >
              Loading...
            </p>
          </div>
        </div>
        <a
          className="better-nav-item "
          href="/recipes"
          onClick={[Function]}
        >
          Browse
        </a>
        <a
          className="better-nav-item "
          href="/schedule/recipes"
          onClick={[Function]}
        >
          Schedule
        </a>
      </div>
    </div>
    <div
      className="c2"
    >
      <img
        alt=""
        className="user-profile-image better-nav-item p-0"
        onClick={[Function]}
        src=""
        tabIndex={0}
      />
      <div
        className="c5"
      >
        <p
          className="bold"
        >
          
        </p>
        <div
          className="d-flex align-center p-1-0"
        >
          <label
            className="d-flex align-items-center cursor-pointer"
          >
            <input
              checked={false}
              className="mr-2"
              onChange={[Function]}
              type="checkbox"
            />
            Dark Mode
          </label>
        </div>
        <p>
          <a
            className="p-1-0"
            href="/settings"
            onClick={[Function]}
          >
            Settings
          </a>
        </p>
        <button
          className="my-button w-100 is-normal"
          onClick={[Function]}
        >
          Logout
        </button>
      </div>
    </div>
  </div>
</nav>
`;

exports[`<Nav/> renders default 1`] = `
.c1 {
  font-size: 1.5rem;
}

.c0 {
  -webkit-flex-wrap: 1;
  -ms-flex-wrap: 1;
  flex-wrap: 1;
  margin-bottom: 0.25rem;
  padding-left: 0.75rem;
  padding-right: 0.75rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

@media (max-width:630px) {
  .c1 {
    display: none;
  }
}

<nav
  className="c0"
>
  <a
    className="better-nav-item pb-1 pt-1 pl-0 pr-0 fw-normal"
    href="/"
    onClick={[Function]}
  >
    <svg
      className="mr-1 fill-text-color "
      version="1.1"
      viewBox="0 0 512 512"
      width="40px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M436.6,87.8c-0.9,0.1-1.9,2.7-3.7,7.9c-2.8,8.1-4.3,14.3-5.2,18.3c-1.4,5.6-3,12.1-5.2,18.8c-2.7,7.9-4.2,12.5-7.7,18.2
        c-4,6.3-8.2,10.6-10.6,13c-2.7,2.7-8,7.5-15.8,12.3c-4.9,3-12.4,6.9-22.3,10c-8.9,2.7-16.9,4.6-24.2,5.9c-2.8,0.5-7.5,1.3-16.4,2.3
        c-5.3,0.6-19.8,2.1-43.9,2.9c-6.7,0.2-15.4,0.4-25.6,0.5c-10.2,0-18.9-0.3-25.6-0.5c-24-0.8-38.5-2.3-43.9-2.9
        c-8.9-1-13.5-1.7-16.4-2.3c-7.3-1.3-15.3-3.2-24.2-5.9c-9.9-3-17.3-6.9-22.3-10c-7.8-4.8-13.1-9.6-15.8-12.3
        c-2.4-2.4-6.6-6.6-10.6-13c-3.6-5.7-5.1-10.3-7.7-18.2c-2.3-6.6-3.8-13.2-5.2-18.8c-1-4-2.4-10.2-5.2-18.3c-1.8-5.2-2.8-7.9-3.7-7.9
        c-4.1-0.3-13.2,23.2-12.1,47.4c0.3,7.5,1.4,25.9,14.4,44.9c10.2,14.8,23.4,23.8,31.6,29.4c10.6,7.2,14.1,7.4,19.7,11.5
        c12.4,9.2,14.8,20.5,22.4,60c6,30.9,11.1,64,22,95.4c2.5,7.3,6.6,27.5,22.5,40.9c7.4,6.2,16.6,10.8,16.6,10.8
        c3.2,1.6,11.6,5.5,23.4,7.8c7.4,1.4,14.1,1.9,20.1,1.8c5.9,0.1,12.7-0.4,20.1-1.8c11.8-2.3,20.2-6.2,23.4-7.8c0,0,9.2-4.6,16.6-10.8
        c15.9-13.4,20-33.6,22.5-40.9c10.9-31.4,16-64.5,22-95.4c7.6-39.6,10.1-50.9,22.4-60c5.6-4.1,9.1-4.3,19.7-11.5
        c8.2-5.6,21.4-14.5,31.6-29.4c13-19,14.1-37.4,14.4-44.9C449.7,110.9,440.6,87.5,436.6,87.8z M333.2,258.7
        c-7.1,53.7-13.8,82-24.6,107.8c-0.4,1.1-1.3,3.3-2.9,6.1c-5.1,9.2-13.1,24.4-31,30.8c-18.3,6.6-36.3,0.3-37.3,0
        c-17.9-6.4-25.9-21.6-31-30.8c-1.6-2.8-2.5-5.1-2.9-6.1c-10.8-25.7-17.5-54.1-24.6-107.8c-1.3-10.1-2.3-18,3.7-22.9
        c6.5-5.3,15.5-4,38.4-3.2c20.4,0.7,49.6,0.7,70,0c23-0.8,31.9-2.1,38.4,3.2C335.4,240.7,334.5,248.6,333.2,258.7z"
      />
    </svg>
    <span
      className="c1"
    >
      Recipe Yak
    </span>
  </a>
  <div
    className="d-flex hide-sm"
  >
    <a
      className="better-nav-item "
      href="/login"
      onClick={[Function]}
    >
      Login
    </a>
    <a
      className="better-nav-item "
      href="/signup"
      onClick={[Function]}
    >
      Signup
    </a>
  </div>
</nav>
`;
